vault/resource_jwt_auth_backend_role.go

```diff
@@ -333,10 +333,12 @@ func jwtAuthBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
-	boundAuds := util.JsonStringArrayToStringArray(resp.Data["bound_audiences"].([]interface{}))
-	err = d.Set("bound_audiences", boundAuds)
-	if err != nil {
-		return fmt.Errorf("error setting bound_audiences in state: %s", err)
+	if resp.Data["bound_audiences"] != nil {
+		boundAuds := util.JsonStringArrayToStringArray(resp.Data["bound_audiences"].([]interface{}))
+		err = d.Set("bound_audiences", boundAuds)
```

Comment:
`d.Set("bound_audiences")` should always be called when reading the Vault role to ensure we detect drift, including if the bound audiences were completely removed:

- https://www.terraform.io/docs/extend/best-practices/detecting-drift.html

If `resp.Data["bound_audiences"]` is `nil` we should probably call `d.Set("bound_audiences",nil)`.

I see this pattern of `if resp.Data["thing"] != nil` in a few places below this as well, which makes me think we may have other issues with possibly missing drift :/

Reply:
Thanks for the review! I agree with your suggestion, but I'd submit that because the issue is present in existing code, it's outside the scope of this change. I can file an issue capturing that, though, if you'd like. Not sure if I'll have the bandwidth to fix it, though.

Reply:
I opened https://github.com/terraform-providers/terraform-provider-vault/issues/767 for the other state issues. 

For this PR, we're potentially introducing drift though because we're no longer always calling `d.Set("bound_audiences", boundAuds)`. For this PR we should add an `else` similar to `oidc_scopes`:

https://github.com/terraform-providers/terraform-provider-vault/blob/7d07921e5bdc0bce7b70f5835dcd61d3b8ad290c/vault/resource_jwt_auth_backend_role.go#L355

What do you think?

Reply:
My personal preference is to fix just the crash bug in this PR, and not the existing drift bug; then go back and fix all the drift problems for this resource (and do an audit for the same problem in other resources; I took a quick look, and most of them seem to do the right thing). But if you'd prefer to see the crash fix and the drift fix for this one attribute to come in together, that's easy enough, and I'm happy to do it.

Reply:
Ah, I think I see your point. In the cases where the code didn't crash, there was no drift, and thus the drift here is a regression. I'll push a fix after lunch.

Reply:
>  there was no drift, and thus the drift here is a regression

Correct! I should have worded my first message to be more clear as to say that was my concern, sorry about that 😄 The change look good!